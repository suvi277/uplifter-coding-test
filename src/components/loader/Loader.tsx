import React from 'react';
import { Spinner } from 'react-bootstrap';
import './Loader.scss';

const Loader: React.FC<{}> = () => (
    <div className="loader">
        <Spinner animation="border" variant="warning" />
    </div>
);

export default Loader;
