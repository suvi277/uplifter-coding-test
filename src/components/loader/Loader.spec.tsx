import React from 'react';
import { shallow } from 'enzyme';
import { Spinner } from 'react-bootstrap';
import Loader from './Loader';

describe('Loader Component', () => {
    const wrapper = shallow(<Loader />);

    it('renders as expected', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should exist loader div', () => {
        expect(wrapper.find('.loader')).toHaveLength(1);
    });

    it('should exist Spinner Component', () => {
        expect(wrapper.exists(Spinner)).toBeTruthy();
    });
});
