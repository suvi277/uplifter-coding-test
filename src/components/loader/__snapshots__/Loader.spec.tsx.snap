// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Component renders as expected 1`] = `ShallowWrapper {}`;
